### PR TITLE
Fix project creation conflict during user creation

### PR DESCRIPTION
**Pull request by rdimitrov**
# Summary

The following PR fixes the conflict that may happen if the project we try to create during `CreateUser` already exists.

The proposed solution checks if there's such project with that name, if there's appends a random 4 char string and tries again until it finds a unique combination.

## Change Type

***Mark the type of change your PR introduces:***

- [x] Bug fix (resolves an issue without affecting existing features)
- [ ] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [ ] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

***Outline how the changes were tested, including steps to reproduce and any relevant configurations. 
Attach screenshots if helpful.***

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [ ] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [ ] Included tests that validate the fix or feature.
- [ ] Checked that related changes are merged.

### CODE REVIEW

File: internal/controlplane/handlers_user.go [hunk 1]
```diff
@@ -78,7 +80,13 @@ func (s *Server) CreateUser(ctx context.Context,
 		// Set up the default project for the user
 		baseName := subject
 		if token.PreferredUsername() != "" {
-			baseName = token.PreferredUsername()
+			// Check if `project_name_lower_idx` unique constraint was violated. This happens when
```
Comment (jhrozek) [inline internal/controlplane/handlers_user.go:83]: Is this an outdated comment? We check the other way around, to see if there's `sql.ErrNoRows` returned right?

File: internal/controlplane/handlers_user.go [hunk 2]
```diff
@@ -483,3 +491,33 @@ func isUserSelfResolving(ctx context.Context, store db.Store, i db.GetInvitation
...
+	if err != nil {
+		return "", err
+	}
+	return hex.EncodeToString(bytes)[:length], nil
+}
+
+// getUniqueProjectBaseName is used to generate a unique project name
+func getUniqueProjectBaseName(ctx context.Context, store db.Store, baseName string) (string, error) {
+	uniqueBaseName := baseName
+	for {
```
Comment (jhrozek) [inline internal/controlplane/handlers_user.go:508]: I know it's unlikely we hit a really infinite loop here, but ` for {}` always makes me nervous. Do you think we could add some upper limit?

**Review by jhrozek — APPROVED** [review]
Since this is a hack to get around an immediate bug, feel free to merge, but please also open an issue for the proper fix

File: internal/controlplane/handlers_user.go (hunk 1 above)
Comment (rdimitrov) [reply internal/controlplane/handlers_user.go:83]: You're right, this was a leftover copy-pasta from the other place where we had this check 👍

File: internal/controlplane/handlers_user.go (hunk 2 above)
Comment (rdimitrov) [reply internal/controlplane/handlers_user.go:508]: I'll add a 10 times retry mechanism. It should be more than enough 👍

**Comment by rdimitrov** [comment]
@jhrozek - Opened an issue to track the proper way to fix this through refactoring - https://github.com/stacklok/minder/issues/3826

**Review by jhrozek — APPROVED** [review]

**Pull request merged**